packages/gasket-plugin-intl/test/build-modules.test.js

```diff
@@ -0,0 +1,235 @@
+/* eslint-disable max-nested-callbacks */
+const assume = require('assume');
+const sinon = require('sinon');
+const fs = require('fs-extra');
+const fsUtils = require('../lib/fs-utils');
+const { BuildModules } = require('../lib/build-modules');
+
+describe('buildModules', function () {
+  const testSrcFilePath = '/path/to/src/myh-fake/locale/en-US.json';
+  const testTgtFilePath = '/path/to/tgt/myh-fake/en-US.json';
+  let logger, mockGasket;
+
+  beforeEach(() => {
+    logger = {
+      log: sinon.stub()
+    };
+    mockGasket = {
+      logger,
+      config: {
+        root: '/path/to/somewhere',
+        intl: {
+          localesDir: '/path/to/somewhere/public/locales',
+          modules: {
+            localesDir: 'locales',
+            excludes: ['cacache', 'yargs', 'axe-core']
+          }
+        }
+      }
+    };
+  });
+
+  describe('BuildModules', function () {
+    let builder;
+
+    beforeEach(function () {
+      builder = new BuildModules(mockGasket);
+    });
+
+    afterEach(function () {
+      builder = null;
+      sinon.restore();
+    });
+
+    it('sets up dir constants', function () {
+      assume(builder).property('_outputDir');
+      assume(builder).property('_nodeModulesDir');
+      assume(builder).property('_lookupDir');
+    });
+
+    describe('#copyFolder', function () {
+      it('calls saveJsonFile for correct data', function (done) {
+        sinon.stub(fs, 'mkdirp').returns(Promise.resolve());
+        sinon.stub(fs, 'readdir').resolves(['test/folder/name.json']);
+        sinon.stub(fsUtils, 'saveJsonFile').callsFake(() => {
+          assume(fs.mkdirp.getCalls()).length(1);
+          assume(fs.readdir.getCalls()).length(1);
+          assume(fs.readFile.getCalls()).length(1);
+          done();
+        });
```

Comment:
why this over having `fsUtils.saveJsonFile` stubbed with `resolves()`, await the `builder.copyFolder` call and move the `assume`s after the await?